### DISCUSSION TITLE
Filter using  date time inequality

### DISCUSSION THREAD

**Discussion by aasutossh** [pocketbase/pocketbase#4846] [Q&A]
I am trying to find the user who are expired. I have this at the moment.
```
      const records = $app.dao().findRecordsByExpr(
        "user_expiration",
        // $dbx.exp("expiration > {:expiration}", {
        //   expiration: new Date(),
        // }),
        $dbx.hashExp({ user: `${user.get("id")}` }),
      );
```
the commented portion doesn't seem to work.
hashExp is fine.
When I console the commented code I simply get empty object.

**Comment by aasutossh** [comment]
consoling the following from the docs also results in empty object.
```
$dbx.exp("total > {:min} AND total < {:max}", { min: 10, max: 30 })
```

**Comment by ganigeorgiev** [comment]
`date` field type values are in the format `Y-m-d H:i:s.uZ` (eg. `2024-05-01 05:30:10.123Z`).
You cannot assign a plain JS Date object as value in the query as it needs to be converted to the correct string format. One way to do it could be `new Date().toISOString().replace("T", " ")` (_replace is needed to convert the date into rfc3339_).

> consoling the following from the docs also results in empty object...

Without information about the data set and your collections structure I don't know how to help. If `total` is not a `number` field type you'll have to [`CAST`](https://www.sqlite.org/lang_expr.html#cast_expressions) it.

If the above doesn't help, please elaborate a little more on your use case and export your collections configuration as json (_Admin UI > Settings > Export collections_).

**Reply by aasutossh** [reply]
will check and let you know. Thank you.
